.changeset/shiny-swans-relax.md

```diff
@@ -0,0 +1,123 @@
+
+// Bad: This loses the schema aware type information. `: TreeNodeSchema` should be omitted to preserve strong typing.
+const schema: TreeNodeSchema = schemaFactory.array(schemaFactory.number);
+const config = new TreeViewConfiguration({ schema });
+
+// This view is types as `TreeView<TreeNodeSchema>`, which does not work well since its missing the actual schema type information.
+const view = tree.viewWith(config);
+// Root is typed as `unknown` allowing invalid assignment operations.
+view.root = "invalid";
+view.root = {};
+// Since all assignments are allowed, valid ones still work:
+view.root = [];
+```
+
+After this change:
+
+
+```typescript
+// Root is now typed as `TreeNode | TreeLeafValue`, still allowing some invalid assignment operations.
+// In the future this should be prevented as well, since the type of the setter in this case should ne `never`
+view.root = "invalid";
+// This no longer compiles:
+view.root = {};
+// This also longer compiles:
+view.root = [];
+```
+
+For code which wants to continue using a unsafe API which can result in runtime errors if the data does not follow the schema, a new alternative has been added to address this use-case. A special type `UnsafeUnknownSchema` can no be used to opt into allowing all valid trees to be provided.
+Note that this leaves ensuring the data is in schema up to the user.
+For now these adjusted APIs can be accessed by casting the view to `TreeViewAlpha<UnsafeUnknownSchema>`.
+If stabilized this option wil be added to `TreeView` directly.
+
+```typescript
+const viewAlpha = view as TreeViewAlpha<UnsafeUnknownSchema>;
+viewAlpha.initialize([]);
+viewAlpha.root = [];
+```
+
+Additionally this seems to have negatively impacted co recursive schema which declare an corecursive array as the first schema in the co-recursive cycle.
+Like the TypeScript language our schema system is built on, we don't guarantee exactly which recursive type will compile, but will do our best to ensure useful recursive schema can be created easily.
+In this case a slight change may be required to some recursive schema to get them to compile again:
+
+For example this schema used to compile:
+
+
+```typescript
+class A extends sf.arrayRecursive("A", [() => B]) {}
+{
+	type _check = ValidateRecursiveSchema<typeof A>;
+}
+// Used to work, but breaks in this update.
+class B extends sf.object("B", { x: A }) {}
+```
+
+But now you must use the recursive functions like `objectRecursive` for types which are co-recursive with an array in some cases.
+In our example, it can be fixed as follows:
+
+```typescript
+class A extends sf.arrayRecursive("A", [() => B]) {}
+{
+	type _check = ValidateRecursiveSchema<typeof A>;
+}
+// Fixed corecursive type, using "Recursive" method variant to declare schema.
+class B extends sf.objectRecursive("B", { x: A }) {}
+{
+	type _check = ValidateRecursiveSchema<typeof B>;
+}
+```
+
+Reversing which type is using the recursive utility also works (shown below).
```

Comment:
Given that the above pattern is our actual recommendation, do we need to note this?

Reply:
This was mostly intended to direct people away from relying on this despite it compiling. We don't want people doing what I did in the past and thing they are clever and getting a better devx by using the recursive helpers less since it might break again but differently.

Reply:
I'm fine with removing it if you think that would be better. I partly just included it since it was interesting and should be discussed in this review.

Reply:
Maybe just rephrase it to say something like "Note, while the following pattern may still compile, we recommend using the previous pattern instead"?